Fixes #1975: Decorate each post with a class that matches the blog host platform

rjayroso opened the pull request:
## Issue This PR Addresses
Fixes #1975 

## Type of Change
- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This essentially decorates `<section class="telescope-post-content">` with the blogging host platform name.  

**For example**, if the URL is `https://roycedev.medium.com/spending-the-weekend-thinking-575bcf535e58` it will extract **only** the `medium-com` and decorate the section class, like so:  `<section class="telescope-post-content medium-com">`  

![image](https://user-images.githubusercontent.com/48869737/112765438-3a150d00-8fdb-11eb-90be-b88e6f470aba.png)  

**Another example:**  
![image](https://user-images.githubusercontent.com/48869737/112765861-36828580-8fdd-11eb-8e0f-f3126360dad0.png)


This will allow greater CSS customization with each post since each blogging platform styles their blogs/layout differently.
In reference to  the BlogSpot issue (#2039), we can fix it by styling **only** BlogSpot images to not be inline.

## How to Test
1. View the changed files for syntax/logic errors and possible changes/suggestions
2. View Vercel deployment OR pull this pr onto your local machine via `gh pr checkout` (see [here](https://cli.github.com/manual/gh_pr_checkout))
3. Scroll down to any blog post, and right-click the *content* (not the title/author name, aim for the paragraphs) and click inspect
![image](https://user-images.githubusercontent.com/48869737/112765658-261ddb00-8fdc-11eb-9d8c-51e435768f03.png)
4. View the `<section>` tag and see the blogging host platform name within the `className` 
![image](https://user-images.githubusercontent.com/48869737/112765691-51a0c580-8fdc-11eb-8fec-78f61b260567.png)


## Checklist
- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

humphd commented on src/web/src/components/Posts/Post.tsx:
```diff
@@ -193,6 +203,10 @@ const PostComponent = ({ postUrl }: Props) => {
   // Grab the post data from our backend so we can render it
   const { data: post, error } = useSWR<Post>(postUrl);
   const [expandHeader, setExpandHeader] = useState(false);
+  // If post exists, extract the blog platform name
```
Move this down to ~248 where you know post exists for sure.  Don't bother if it doesn't.  Also, I would suggest you should do it all in a function that you call vs. using a variable:

This needs a space between.  I'd do it like this:

```jsx
className={`telescope-post-content ${extractBlogClassName(post.url)}`}
```

humphd commented on src/web/src/components/Posts/Post.tsx:
```diff
@@ -184,6 +184,16 @@ const formatPublishedDate = (dateString: string) => {
 };
 
+const parseHostName = (hostname: string) => {
+  // Check if there is a prefix to the hostname
```
I think you can simplify this a bit.  You are trying to deal with all possible cases, but we only have a handful that we care about (medium, dev.to, etc).  You can hard code them in here and check if the hostname ends with those strings:

```js
if(hostname.endsWith('medium.com')) {
  return "is-medium";
}
if(hostname.endsWith('dev.to')) {
  return "is-devto";
}
...
// Maybe add a "generic" version for blogs we don't know about specifically...
return 'is-generic';
```

const knownHosts = ['medium.com', 'dev.to', ...];
return 
```

humphd commented on src/web/src/components/Posts/Post.tsx:
```diff
@@ -284,7 +298,7 @@ const PostComponent = ({ postUrl }: Props) => {
         <section
           ref={sectionEl}
-          className="telescope-post-content"
+          className={'telescope-post-content' + blogClassName}
```
See my comment above.  I would call a function here that returns the string you want.

birtony reviewed (dismissed):
I tend to agree with @humphd's comments. It only really makes sense to add a platform-specific `className` if we know about the platform and apply appropriate styles. Otherwise, even with distinctive `className`s all of the posts will be identical for us, since we don't apply any specific styles. We can extend the list of the known platforms when needed and add corresponding styles.

rjayroso commented:
Implemented the changes suggested above to the best of my understanding.
1. Removed variable declaration/function call for `blogClassName` and combined it within the `className` like so `className={'telescope-post-content ${extractBlogClassName(new URL(post.url).hostname)}'}` 
2. `parseHostName` function changed to `extractBlogClassName` (we only care about the known hosts, otherwise label it as generic)

**Intended functionality is still preserved:**
![Capture](https://user-images.githubusercontent.com/48869737/113463629-ffeda600-93f4-11eb-8ecb-66d2c97d7f71.PNG)
![Capture2](https://user-images.githubusercontent.com/48869737/113463630-011ed300-93f5-11eb-8e60-de91184b182b.PNG)

humphd commented on src/web/src/components/Posts/Post.tsx:
```diff
@@ -284,7 +297,7 @@ const PostComponent = ({ postUrl }: Props) => {
         <section
           ref={sectionEl}
-          className="telescope-post-content"
+          className={`telescope-post-content ${extractBlogClassName(new URL(post.url).hostname)}`}
```
let's move the `new URL(post.url).hostname)` bit into `extractBlogClassName` and just pass the `post.url` here.

rjayroso replied:
Is something like this what you had in mind?
```js
const extractBlogClassName = (url: string) => {
  const blogClassName = new URL(url).hostname;
  if (blogClassName.endsWith('medium.com')) {
    return 'is-medium';
  } 
  // ... rest of function ...
}
```
This is what I pushed just now, I've checked and the class name decoration still works as intended

humphd reviewed (changes requested):
Nice.  One more change and I think this is good.

HyperTHD reviewed (commented):
Is this for specific blog posting websites? Cause wordpress is also used alot too.

EDIT: I read the comments above, this is GTM, just make sure you rebase first
